<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AttributeBasedClusteringDialog</class>
 <widget class="QDialog" name="AttributeBasedClusteringDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>549</width>
    <height>366</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Attribute based clustering</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="1" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Fields:</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1" colspan="2">
    <widget class="QPushButton" name="addFieldButton">
     <property name="text">
      <string>---&gt;</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1" colspan="2">
    <widget class="QPushButton" name="deleteAllButton">
     <property name="text">
      <string>Delete all</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1" colspan="2">
    <widget class="QPushButton" name="deleteSelectedButton">
     <property name="text">
      <string>Delete selected</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="3">
    <widget class="QProgressBar" name="progressBar">
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="value">
      <number>-1</number>
     </property>
    </widget>
   </item>
   <item row="6" column="4">
    <widget class="QPushButton" name="runButton">
     <property name="text">
      <string>Run</string>
     </property>
    </widget>
   </item>
   <item row="6" column="5">
    <widget class="QPushButton" name="cancelButton">
     <property name="text">
      <string>Cancel</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Vector layer:</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1" colspan="2">
    <widget class="QgsFieldComboBox" name="fieldsComboBox"/>
   </item>
   <item row="0" column="1" colspan="2">
    <widget class="QgsMapLayerComboBox" name="vectorLayerComboBox"/>
   </item>
   <item row="5" column="0" colspan="6">
    <widget class="QGroupBox" name="settingsGroup">
     <property name="title">
      <string>Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="4" column="2">
       <widget class="QComboBox" name="linkageMethodComboBox">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <item>
         <property name="text">
          <string>single</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>complete</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>average</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>median</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>weighted</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ward</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QLineEdit" name="outputFieldNameLine">
        <property name="text">
         <string>class</string>
        </property>
        <property name="maxLength">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QComboBox" name="clusteringMethodComboBox">
        <item>
         <property name="text">
          <string>K-Means (not weighted, scipy required, fast, known number of clusters)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Hierarchical (weighted, native, slow, known number of clusters)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Hierarchical-2 (not weighted, scipy required, fast, auto number of clusters)</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Number of clusters</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Linkage method</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QSpinBox" name="numberOfClustersSpinBox">
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLineEdit" name="clusteringThresholdLine">
        <property name="text">
         <string>0.5</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Clusteng method</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Clustering threshold</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Output field name</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QCheckBox" name="checkBox">
        <property name="text">
         <string>Normalize attributes</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="3">
    <widget class="QLabel" name="helpLink">
     <property name="text">
      <string>&lt;a href=&quot;http://www.ekazakov.info/projects/abc_tutorial.html&quot;&gt;Help&lt;/a&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="openExternalLinks">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="3" rowspan="5" colspan="3">
    <widget class="QTableWidget" name="fieldsTable">
     <column>
      <property name="text">
       <string>Field</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Weight</string>
      </property>
     </column>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsFieldComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsfieldcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
